client/db/src/storage_cache.rs

```diff
@@ -495,19 +499,25 @@ impl<S: StateBackend<HasherFor<B>>, B: BlockT> StateBackend<HasherFor<B>> for Ca
 		// Note that local cache makes that lru is not refreshed
 		if let Some(entry) = local_cache.storage.get(key).cloned() {
 			trace!("Found in local cache: {:?}", HexDisplay::from(&key));
-			return Ok(entry)
+			return Ok(
+				self.usage.tally_key_read(key, entry, true)
+			)
 		}
 		let mut cache = self.cache.shared_cache.lock();
 		if Self::is_allowed(Some(key), None, &self.cache.parent_hash, &cache.modifications) {
 			if let Some(entry) = cache.lru_storage.get(key).map(|a| a.clone()) {
 				trace!("Found in shared cache: {:?}", HexDisplay::from(&key));
-				return Ok(entry)
+				return Ok(
+					self.usage.tally_key_read(key, entry, true)
+				)
```

Comment:
This looks a it strange to me, I would prefer that you do the tally and then return `Ok(entry)`. Same with others.

Reply:
It just passes value through counter, why strange?

Reply:
It's just a nit :). Because the return value of the function doesn't depend on `tally_key_read` as it looks. And a reader (like me) may be forced to check the definition unnecessarily. Also, I think is not very stylistic to write `Ok(func(..))`.